app/merchant/views.py

```diff
@@ -116,6 +125,46 @@ def add_to_cart():
     return redirect(url_for('.listing_view_all'))
 
 
+@merchant.route('/add_quantity_to_cart')
+@login_required
+@merchant_required
+def add_quantity_to_cart():
+    listing_id = request.args.get('listing_id')
+    quantity = int(request.args.get('quantity'))
+    return_to = request.args.get('return_to')
+    if not quantity:
+        quantity = 0
+
+    if not listing_id:
+        return redirect(url_for('.index'))
+    listing = Listing.query.filter_by(id=listing_id).first()
+    cart_item = CartItem.query.filter_by(
+        merchant_id=current_user.id).filter_by(
+        listing_id=listing_id).first()
+
+    if listing is None:
+        abort(404)
+
+    if cart_item is not None:
+        db.session.delete(cart_item)
+        db.session.commit()
+
+    if quantity >= 1:
+        db.session.add(CartItem(
+            merchant_id=current_user.id,
+            listing_id=listing_id,
+            quantity=quantity
+        ))
+        db.session.commit()
+
+    if return_to == 'listing_view_all':
```

Comment:
When we combine this function with `change_in_cart`, we can also get rid of the redirects and `return_to`s, since we'll just be using AJAX and sending a response back to the client